pkg/ipam/types.go

```diff
@@ -12,24 +12,19 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package container
+package ipam
 
 import (
 	"github.com/cilium/cilium/pkg/lock"
 
-	dTypes "github.com/docker/engine-api/types"
+	"github.com/containernetworking/cni/plugins/ipam/host-local/backend/allocator"
+	"k8s.io/kubernetes/pkg/registry/core/service/ipallocator"
 )
 
-type Container struct {
-	// Mutex internal mutex for the whole container structure
-	Mutex lock.RWMutex
-	dTypes.ContainerJSON
-}
-
-// NewContainer a Container with its labels initialized.
-func NewContainer(dc *dTypes.ContainerJSON) *Container {
-	// FIXME should we calculate LabelsHash here?
-	return &Container{
-		ContainerJSON: *dc,
-	}
+// Config is the IPAM configuration used for a particular IPAM type.
+type Config struct {
+	IPAMConfig     allocator.IPAMConfig
+	IPv6Allocator  *ipallocator.Range
+	IPv4Allocator  *ipallocator.Range
+	AllocatorMutex lock.RWMutex
```

Comment:
- Can't the mutex be private?

- Can you put the allocator in the line(s) above the parameters that is protecting?

Reply:
made it private. the mutex protects the entire struct. added a comment.